Tasks/Gradle/sonar.gradle

```diff
@@ -0,0 +1,26 @@
+allprojects {
+    // Apply to the main build script a new repository and a dependency on the SonarQube plugin
+    buildscript  {
+        repositories {
+            maven {
+                url "https://plugins.gradle.org/m2/"
+            }
+        }
+        dependencies {
+            classpath "org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.0.1"
+        }
+    }
+
+    // After the project has been evaluated (and the dependency above has been added, apply the SonarQube plugin
+    afterEvaluate { project ->
+        if (project.getParent() == null && !project.plugins.hasPlugin('org.sonarqube')) {
+            //Add sonar plugin only to root projects, and only if it is not already applied
+            project.apply plugin: 'org.sonarqube'
+        }
+
+        // Warn if the project name has not been set
+        if (project.name.equalsIgnoreCase("s")) {
+            logger.warn('Project name is not set.')
```

Comment:
Although this uses Gradle's standard SL4J warning message method, this does not show up any different from normal stdout in the logs.